File: NetFramework/Ribbon/CS/WordPad/BackstageTabs/Items/ItemButton.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace WordPad.BackstageTabs.Items
{
    public partial class ItemButton : UserControl
    {
        #region private fields
        private Color _defaultBackColor = Color.FromArgb(243, 243, 243);
        private Color _hotBackColor = Color.FromArgb(198, 198, 198);
        private Color _hotPressedBackColor = Color.FromArgb(195, 195, 195);
        #endregion

        #region ctor
        public ItemButton()
        {
            InitializeComponent();
            BackColor = _defaultBackColor;
        }
        #endregion

        #region properties
        public Image ItemImage
        {
            get { return pbImage.Image; }
            set { pbImage.Image = value; }
        }

        public string ItemText
        {
            get { return lblText.Text; }
            set { lblText.Text = value; }
        }
        #endregion

        #region event handlers
        private void MenuItem_MouseDown(object sender, MouseEventArgs e)
        {
            BackColor = _hotPressedBackColor;
        }

        private void MenuItem_MouseUp(object sender, MouseEventArgs e)
        {
            BackColor = _hotBackColor;
        }

        private void MenuItem_MouseLeave(object sender, EventArgs e)
        {
            BackColor = _defaultBackColor;
        }

        private void MenuItem_MouseEnter(object sender, EventArgs e)
        {
            BackColor = _hotBackColor;
        }

        private void MenuItem_Click(object sender, EventArgs e)
        {
            OnClick(e);
        }
        #endregion

    }
}

File: NetFramework/Ribbon/VB/WordPad/BackstageTabs/Items/ItemButton.vb
Public Class ItemButton

    Private _defaultBackColor As Color = Color.FromArgb(243, 243, 243)
    Private _hotBackColor As Color = Color.FromArgb(198, 198, 198)
    Private _hotPressedBackColor As Color = Color.FromArgb(195, 195, 195)

    Public Property ItemImage As Image
        Get
            Return pbImage.Image
        End Get
        Set(ByVal value As Image)
            pbImage.Image = value
        End Set
    End Property

    Public Property ItemText As String
        Get
            Return lblText.Text
        End Get
        Set(ByVal value As String)
            lblText.Text = value
        End Set
    End Property

    Private Sub MenuItem_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles MyBase.MouseDown, pbImage.MouseDown, lblText.MouseDown
        BackColor = _hotPressedBackColor
    End Sub

    Private Sub MenuItem_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs) Handles MyBase.MouseUp, pbImage.MouseUp, lblText.MouseUp
        BackColor = _hotBackColor
    End Sub

    Private Sub MenuItem_MouseLeave(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.MouseLeave, pbImage.MouseLeave, lblText.MouseLeave
        BackColor = _defaultBackColor
    End Sub

    Private Sub MenuItem_MouseEnter(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.MouseEnter, pbImage.MouseEnter, lblText.MouseEnter
        BackColor = _hotBackColor
    End Sub

    Private Sub MenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles pbImage.Click, lblText.Click
        Me.OnClick(EventArgs.Empty)
    End Sub

    Private Sub ItemButton_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        BackColor = _defaultBackColor
    End Sub
End Class
